lib/wallet/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -3992,6 +4067,74 @@ prop_balanceTransactionExistingReturnCollateral
             Left err -> ErrBalanceTxExistingReturnCollateral === err
             e -> counterexample (show e) False
 
+prop_bootstrapWitnesses
+    :: (Word8 -> WriteTx.InAnyRecentEra Cardano.Tx -> Property)
+    -> Word8
+    -- ^ Number of bootstrap witnesses.
+    --
+    -- Testing with [0, 255] should be sufficient.
+    -> AnyRecentEra
+    -> Cardano.NetworkId
+    -- ^ Network - will be encoded inside the witness.
+    -> Index 'WholeDomain 'AccountK
+    -- ^ Account index - will be encoded inside the witness.
+    -> Index 'WholeDomain 'CredFromKeyK
+    -- ^ Index for the first of the 'n' addresses.
+    -> Property
+prop_bootstrapWitnesses
+    p n (AnyRecentEra (era :: RecentEra era)) net accIx addr0Ix =
+    let
+        -- Start incrementing the ixs upward, and if we reach 'maxBound', loop
+        -- around, to ensure we always have 'n' unique indices.
+        addrIxs = take (fromIntegral n)
+            $ [addr0Ix .. maxBound] ++ [minBound .. addr0Ix]
+        body = emptyCardanoTxBody
+        wits :: [Cardano.KeyWitness era]
+        wits = map (dummyWitForIx body) addrIxs
+    in
+        p n (WriteTx.InAnyRecentEra era $ Cardano.Tx body wits)
+  where
+    emptyCardanoTxBody = body
+      where
+        Cardano.Tx body _ = WriteTx.toCardanoTx @era $ WriteTx.emptyTx era
+
+    mw = SomeMnemonic $ either (error . show) id
+        (entropyToMnemonic @12 <$> mkEntropy "0000000000000000")
```

Comment:
I noticed that this definition is already repeated in four different places within `TransactionSpec`.

Perhaps we could factor out the repetition, i.e., by making a top-level dummy mnemonic?

```hs
dummyMnemonic = SomeMnemonic $ either (error . show) id
    (entropyToMnemonic @12 <$> mkEntropy "0000000000000000")
```

Reply:
Done